### DISCUSSION TITLE
path with case insentive breaks routing table

### DISCUSSION THREAD

**Discussion by flowdopip** [Kong/kong#9578] [Help] [answered]
Hi all,

I'm trying to adopt a legacy API that has the following segregation:

## Upstream A
- ~/(?i)ext/content 

## Upstream B
- ~/(?i)ext/content/v1/search/contents
- ~/(?i)ext/content/v1/spaces
- ~/(?i)ext/content/v1/modules

## Upstream C
- ~/(?i)ext/content/v1/zones
- ~/(?i)ext/content/v1/countries
- ~/(?i)ext/content/v1/countryproperties
- ~/(?i)ext/content/v1/whitelists
- ~/(?i)ext/content/v1/zonetypes

If the service A (upstream A) has the case insensitive regex the rules do not work ( all paths ), but if I remove ~/(?i) (/ext/content ) everything works as expected.

I have a postman collection to run the tests, and over Kong 3.0.0 the results are consistent, but applying the same condition on kong 2.8.1 the results of test are different for every time that I boot the docker solution.

My kong.yml
`_format_version: "3.0"
_comment: "Global Plugins"
services:
- name: ext-content-seo
  url: http://echo-server/
  protocol:
  - https
  connect_timeout: 100000
  read_timeout: 100000
  write_timeout: 100000
  retries: 3
  routes:
  - name: ext-content-seo-fallback
    strip_path: true
    hosts:
    - my.host.net
    paths:
    - /ext/content
    methods:
    - GET
    - POST
    - PUT
    - PATCH
    - DELETE
    - OPTIONS
    - HEAD
    protocols:
    - http
- name: ext-content-content
  url: http://echo-server/
  protocol:
  - https
  connect_timeout: 100000
  read_timeout: 100000
  write_timeout: 100000
  retries: 3
  routes:
  - name: ext-content-content-fallback
    strip_path: false
    hosts:
    - my.host.net
    paths:
    - ~/(?i)ext/content/v1/search/contents
    - ~/(?i)ext/content/v1/spaces
    - ~/(?i)ext/content/v1/modules
    methods:
    - GET
    - POST
    - PUT
    - PATCH
    - DELETE
    - OPTIONS
    - HEAD
    protocols:
    - http
- name: ext-content-core
  url: http://echo-server/
  protocol:
  - https
  connect_timeout: 100000
  read_timeout: 100000
  write_timeout: 100000
  retries: 3
  routes:
  - name: ext-content-core-fallback
    strip_path: false
    hosts:
    - my.host.net
    paths:
    - ~/(?i)ext/content/v1/zones
    - ~/(?i)ext/content/v1/countries
    - ~/(?i)ext/content/v1/countryproperties
    - ~/(?i)ext/content/v1/whitelists
    - ~/(?i)ext/content/v1/zonetypes
    methods:
    - GET
    - POST
    - PUT
    - PATCH
    - DELETE
    - OPTIONS
    - HEAD
    protocols:
    - http`

Im doing test base on Kong-Debug: 1 header results.

**Comment by hanshuebner** [comment, marked as answer]
Hello @flowdopip,

sorry for the very long delay.  The problem that you observe is caused by the fact that you are using regular expression routes, and that there is not automatic priorization of routes based on the length of the regular expression or the length of the match in that case.  It is thus not defined whether your "Upstream A" has a higher or lower priority value than "Upstream B" or "Upstream C".  You can enforce an explcit order among regular expression routes using the `regex_priority` attribute.  Please refer to the [documentation on routes](https://docs.konghq.com/gateway/latest/key-concepts/routes/) for more information.

Kind regards,
Hans
